src/common/xo-week-heatmap/index.js

```diff
@@ -0,0 +1,172 @@
+import React from 'react'
+import _ from 'messages'
+import forEach from 'lodash/forEach'
+import map from 'lodash/map'
+import maxBy from 'lodash/maxBy'
+import minBy from 'lodash/minBy'
+import times from 'lodash/times'
+import { injectIntl } from 'react-intl'
+
+import Component from '../base-component'
+import Tooltip from '../tooltip'
+import { propTypes } from '../utils'
+
+import styles from './index.css'
+
+const TIME_FORMAT = {
+  day: 'numeric',
+  hour: 'numeric',
+  minute: 'numeric',
+  month: 'short',
+  second: 'numeric',
+  weekday: 'short',
+  year: 'numeric'
+}
+
+// ===================================================================
+
+@propTypes({
+  cellRenderer: propTypes.func,
+  colors: propTypes.arrayOf(
+    propTypes.string
+  ),
+  data: propTypes.arrayOf(
+    propTypes.shape({
+      date: propTypes.number.isRequired,
+      value: propTypes.number.isRequired
+    })
+  ).isRequired
+})
+@injectIntl
+export default class XoWeekHeatmap extends Component {
+  static defaultProps = {
+    cellRenderer: value => value,
+    colors: [
+      '#ffffee',
+      '#ffffd9',
+      '#edf8b1',
+      '#c7e9b4',
+      '#7fcdbb',
+      '#41b6c4',
+      '#1d91c0',
+      '#225ea8',
+      '#253494',
+      '#081d58'
+    ]
+  }
+
+  componentWillReceiveProps = nextProps => this._updateState(nextProps)
+  componentWillMount = this._updateState
+
+  _computeIntervals = ({ colors, data }) => {
+    const min = minBy(data, 'value').value
+    const max = maxBy(data, 'value').value
```

Comment:
You can also use [`d3.extent`](https://github.com/d3/d3-array#extent) if you want.